### PR TITLE
feat: Extra Annotations - Add support for resource annotations

### PR DESCRIPTION
**Pull request by jccastillocano**
## What

Adding `annotations: {}` value for including custom annotations on the following resources:

- Secret
- ConfigMap
- Deployment (admin)
- CronJob
- Job

Updated doc with new values plus fixed some missing value descriptions.

## Why

There is limited support for annotations in the chart resources, only at the pod level (for both admin deployment and cronjob) as well as service account (custom annotations).

## Tests

Tested with the following scenarios:

### Without annotations

Using default value (empty dict) and rendering the chart with `helm template cloudquery . -f values.yaml --debug` it produces all resources without annotations:

```yaml
---
# Source: cloudquery/templates/secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: cloudquery-secret
  labels:
    helm.sh/chart: cloudquery-31.1.0
    app.kubernetes.io/name: cloudquery
    app.kubernetes.io/instance: cloudquery
    app.kubernetes.io/version: "5.1"
    app.kubernetes.io/managed-by: Helm
type: Opaque
...
---
# Source: cloudquery/templates/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: cloudquery-config
  labels:
    helm.sh/chart: cloudquery-31.1.0
    app.kubernetes.io/name: cloudquery
    app.kubernetes.io/instance: cloudquery
    app.kubernetes.io/version: "5.1"
    app.kubernetes.io/managed-by: Helm
data:
...
---
# Source: cloudquery/templates/admin-deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: cloudquery-admin
  labels:
    helm.sh/chart: cloudquery-31.1.0
    app.kubernetes.io/name: cloudquery
    app.kubernetes.io/instance: cloudquery
    app.kubernetes.io/version: "5.1"
    app.kubernetes.io/managed-by: Helm
spec:
...
---
# Source: cloudquery/templates/cronjob.yaml
apiVersion: batch/v1
kind: CronJob
metadata:
  name: cloudquery-cron
  labels:
    helm.sh/chart: cloudquery-31.1.0
    app.kubernetes.io/name: cloudquery
    app.kubernetes.io/instance: cloudquery
    app.kubernetes.io/version: "5.1"
    app.kubernetes.io/managed-by: Helm
spec:
...
```

### With custom annotations

Having `test_values.yaml` as follows:

```yaml
annotations:
  service: cloudquery
  url: https://cloudquery.io
```

Rendering the chart with `helm template cloudquery . -f values.yaml -f test_values.yaml --debug` it produces the following output, which includes custom annotations specified in `test_values.yaml`:

```yaml
---
# Source: cloudquery/templates/secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: cloudquery-secret
  labels:
    helm.sh/chart: cloudquery-31.1.0
    app.kubernetes.io/name: cloudquery
    app.kubernetes.io/instance: cloudquery
    app.kubernetes.io/version: "5.1"
    app.kubernetes.io/managed-by: Helm
  annotations:
    service: "cloudquery"
    url: "https://cloudquery.io"
...
---
# Source: cloudquery/templates/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: cloudquery-config
  labels:
    helm.sh/chart: cloudquery-31.1.0
    app.kubernetes.io/name: cloudquery
    app.kubernetes.io/instance: cloudquery
    app.kubernetes.io/version: "5.1"
    app.kubernetes.io/managed-by: Helm
  annotations:
    service: "cloudquery"
    url: "https://cloudquery.io"
data:
...
---
# Source: cloudquery/templates/admin-deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: cloudquery-admin
  labels:
    helm.sh/chart: cloudquery-31.1.0
    app.kubernetes.io/name: cloudquery
    app.kubernetes.io/instance: cloudquery
    app.kubernetes.io/version: "5.1"
    app.kubernetes.io/managed-by: Helm
  annotations:
    service: "cloudquery"
    url: "https://cloudquery.io"
spec:
...
---
# Source: cloudquery/templates/cronjob.yaml
apiVersion: batch/v1
kind: CronJob
metadata:
  name: cloudquery-cron
  labels:
    helm.sh/chart: cloudquery-31.1.0
    app.kubernetes.io/name: cloudquery
    app.kubernetes.io/instance: cloudquery
    app.kubernetes.io/version: "5.1"
    app.kubernetes.io/managed-by: Helm
  annotations:
    service: "cloudquery"
    url: "https://cloudquery.io"
spec:
...
```

### CODE REVIEW

**Review by erezrokah — COMMENTED** [review]
Hi @jccastillocano, thanks for the PR 🚀 

Looks like the CI is failing since we're using a different version of `helm-docs`. I think you can either use the same version or simply update https://github.com/cloudquery/helm-charts/blob/f29e2db54016680db980738290cab6268c64bbee/.github/workflows/lint_test.yml#L36

**Comment by jccastillocano** [comment]
> Hi @jccastillocano, thanks for the PR 🚀
> 
> Looks like the CI is failing since we're using a different version of `helm-docs`. I think you can either use the same version or simply update
> 
> https://github.com/cloudquery/helm-charts/blob/f29e2db54016680db980738290cab6268c64bbee/.github/workflows/lint_test.yml#L36

On it, thx!

File: charts/cloudquery/templates/_helpers.tpl [hunk 1]
```diff
@@ -45,6 +45,18 @@ app.kubernetes.io/managed-by: {{ .Release.Service }}
...
 {{- end }}
 
+{{/*
+Common annotations
+*/}}
+{{- define "cloudquery.annotations" -}}
+{{- if .Values.annotations }}
+  annotations:
+    {{- range $key, $value := .Values.annotations }}
+    {{ $key }}: {{ $value | quote }}
```
[the remark this answers is not in the document]
Comment (jccastillocano) [reply charts/cloudquery/templates/_helpers.tpl]: Makes sense, thx for the feedback! Already changed.

**Review by mnorbury — APPROVED** [review]
I ran your test sequence and it looks good to me. Great work.

**Pull request merged**